File: app/src/main/java/com/gmail/cristiandeives/motofretado/TrackBusModel.java
package com.gmail.cristiandeives.motofretado;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.UiThread;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;
import com.gmail.cristiandeives.motofretado.http.Bus;
import com.gmail.cristiandeives.motofretado.http.GetBusesRequest;
import com.gmail.cristiandeives.motofretado.http.GetBusesResponseListener;
import com.gmail.cristiandeives.motofretado.http.ModelListener;
import com.gmail.cristiandeives.motofretado.http.PostBusRequest;
import com.gmail.cristiandeives.motofretado.http.PostBusResponseListener;

import org.json.JSONObject;

@UiThread
class TrackBusModel implements TrackBusMvp.Model {
    private final @NonNull RequestQueue mQueue;

    TrackBusModel(@NonNull Context context) {
        mQueue = Volley.newRequestQueue(context);
    }

    @Override
    public void readAllBuses(@NonNull ModelListener<Bus[]> listener) {
        Request<JSONObject> req = new GetBusesRequest(new GetBusesResponseListener(listener));
        mQueue.add(req);
    }

    @Override
    public void createBus(@NonNull Bus bus, ModelListener<Bus> listener) {
        Request<JSONObject> req = new PostBusRequest(bus, new PostBusResponseListener(listener));
        mQueue.add(req);
    }

    @Override
    public void cancelAllRequests() {
        mQueue.cancelAll(req -> true);
    }
}

File: app/src/main/java/com/gmail/cristiandeives/motofretado/TrackBusModel.kt
package com.gmail.cristiandeives.motofretado

import android.content.Context
import android.support.annotation.UiThread
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley
import com.gmail.cristiandeives.motofretado.http.Bus
import com.gmail.cristiandeives.motofretado.http.BusResponseListener
import com.gmail.cristiandeives.motofretado.http.BusesResponseListener
import com.gmail.cristiandeives.motofretado.http.GetBusesRequest
import com.gmail.cristiandeives.motofretado.http.ModelListener
import com.gmail.cristiandeives.motofretado.http.PostBusRequest

@UiThread
internal class TrackBusModel(context: Context) : TrackBusMvp.Model {
    private val mQueue: RequestQueue = Volley.newRequestQueue(context)

    override fun readAllBuses(listener: ModelListener<List<Bus>>) {
        val req = GetBusesRequest(BusesResponseListener(listener))
        mQueue.add(req)
    }

    override fun createBus(bus: Bus, listener: ModelListener<Bus>) {
        val req = PostBusRequest(bus, BusResponseListener(listener))
        mQueue.add(req)
    }

    override fun cancelAllRequests() {
        mQueue.cancelAll { true }
    }
}